### PR TITLE
Added dataset selection into examples/self_supervised

### PR DESCRIPTION
**Pull request by Nimrais**
## Before submitting (checklist)

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [ ] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [x] Did you check that your code passes the unit tests `pytest .` ? 
- [ ] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?
- [ ] Did you run [colab minimal CI/CD](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/colab_ci_cd.ipynb) with `latest` and `minimal` requirements?
- [ ] Did you check XLA integration with [single](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_single_process.ipynb) and [multiple](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_multi_process.ipynb) processes?

## Description

Add CIFAR-100 and STL-10 for testing in the catalyst self-supervised example.

## Type of Change

- [x] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### FAQ
Please review the FAQ before submitting an issue:
- [x] I have read the [documentation and FAQ](https://catalyst-team.github.io/catalyst/)
- [x] I have reviewed the [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [x] I have checked the [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [x] have read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [x] I have joined [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

### CODE REVIEW

File: examples/self_supervised/barlow_twins.py [hunk 1]
```diff
@@ -56,6 +57,10 @@ def forward(self, x):
 parser.add_argument(
     "--logdir", default="./logdir", type=str, help="Logs directory (tensorboard, weights, etc)"
 )
+parser.add_argument(
+    "--dataset", default="STL10", type=str, help="Dataset: CIFAR-10, CIFAR-100 or STL10"
```
Comment (Scitator) [inline examples/self_supervised/barlow_twins.py]:
```suggestion
    "--dataset", default="STL10", type=str, choices=datasets.keys(), help="Dataset: CIFAR-10, CIFAR-100 or STL10"
```

**Review by Scitator — DISMISSED** [review]

File: examples/self_supervised/barlow_twins.py [hunk 2]
```diff
@@ -14,13 +14,16 @@
 from catalyst.contrib.models.cv.encoders import ResnetEncoder
 from catalyst.contrib.nn import BarlowTwinsLoss
 from catalyst.data import SelfSupervisedDatasetWrapper
+from examples.self_supervised.common import add_arguments, datasets
```
Comment (Scitator) [inline examples/self_supervised/barlow_twins.py]:
```suggestion
from .common import add_arguments, datasets
```
should it work?

File: examples/self_supervised/common.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+from datasets import datasets
+
+
+def add_arguments(parser):
```
Comment (Scitator) [inline examples/self_supervised/common.py]: add here a one-line description please :)

Comment (Nimrais) [reply examples/self_supervised/common.py]: [Done](https://github.com/catalyst-team/catalyst/pull/1300/commits/8a04def4902189a987f1639e4d09cda899964c2b)?

**Pull request merged**